client/src/components/Image_components/ProductImages/ProductImages.jsx

```diff
@@ -53,6 +55,41 @@ export default class ProductImages extends React.Component {
     });
   }
 
+  handlePrevClick() {
+    const { currentIndex } = this.state;
+    const { cardList } = this.props;
+    let index = currentIndex;
+
+    if (index < 4) {
+      index = cardList.length - 1;
+      document.getElementById('carouselComponent').scrollBy(1260, 0);
+    } else {
+      index -= 1;
+      document.getElementById('carouselComponent').scrollBy(-126, 0);
+    }
+    this.setState({
+      currentIndex: index,
+    });
+  }
+
+  handleNextClick() {
+    const { currentIndex } = this.state;
+    const { cardList } = this.props;
+    let index = currentIndex;
+
+    if (index === cardList.length - 4) {
+      index = 0;
+      document.getElementById('carouselComponent').scrollBy(-1260, 0);
+    } else {
+      index += 1;
+      document.getElementById('carouselComponent').scrollBy(126, 0);
+    }
+
+    this.setState({
+      currentIndex: index,
+    });
```

Comment:
Looks like you are refining the carousel behavior. I look forward to trying it live.